test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -0,0 +1,299 @@
+        assert !columns.isEmpty() : "Unable to find partition key columns";
+        if (columns.size() == 1)
+            return getTypeSupport(columns.get(0).type).bytesGen();
+        List<Gen<ByteBuffer>> columnGens = new ArrayList<>(columns.size());
+        for (ColumnMetadata cm : columns)
+            columnGens.add(getTypeSupport(cm.type).bytesGen());
+        return rnd -> {
+            ByteBuffer[] buffers = new ByteBuffer[columnGens.size()];
+            for (int i = 0; i < columnGens.size(); i++)
+                buffers[i] = columnGens.get(i).generate(rnd);
+            return CompositeType.build(buffers);
+        };
+    }
+
+    /**
+     * Hacky workaround to make sure different generic MessageOut types can be used for {@link #MESSAGE_GEN}.
+     */
+    private static final Gen<Message<?>> cast(Gen<? extends Message<?>> gen)
+    {
+        return (Gen<Message<?>>) gen;
+    }
+
+    /**
+     * Java's type inferrence with chaining doesn't work well, so this is used to inferr the root type early in cases
+     * where javac can't figure it out
+     */
+    private static <T> Gen<T> gen(Gen<T> fn)
+    {
+        return fn;
+    }
+
+    /**
+     * Converts the type to a CQL type.  This method special cases empty and UDTs so the string can be used in a create
+     * statement.
+     *
+     * Special cases
+     * <ul>
+     *     <li>empty - replaces with 'org.apache.cassandra.db.marshal.EmptyType'.  empty is the tostring of the type in
+     *     CQL but not allowed to create as empty, but fully qualified name is allowed</li>
+     *     <li>UserType - replaces with TupleType</li>
+     * </ul>
+     */
+    public static String toCqlType(AbstractType<?> type)
+    {
+        return type.expandUserTypes().asCQL3Type().toString().replace("empty", "'org.apache.cassandra.db.marshal.EmptyType'");
+    }
+
+    /**
+     * Uses reflection to generate a toString.  This method is aware of common Cassandra classes and can be used for
+     * generators or tests to provide more details for debugging.
+     */
+    public static String toStringRecursive(Object o)
+    {
+        return ReflectionToStringBuilder.toString(o, new MultilineRecursiveToStringStyle()
+        {
+            private String spacer = "";
+
+            {
+                // common lang uses start/end chars that are not the common ones used, so switch to the common ones
+                setArrayStart("[");
+                setArrayEnd("]");
+                setContentStart("{");
+                setContentEnd("}");
+                setUseIdentityHashCode(false);
+                setUseShortClassName(true);
+            }
+
+            protected boolean accept(Class<?> clazz)
+            {
+                return !clazz.isEnum() // toString enums
+                       && Stream.of(clazz.getDeclaredFields()).filter(f -> !Modifier.isStatic(f.getModifiers())).findAny().isPresent(); // if no fields, just toString
```

Comment:
```suggestion
                       && Stream.of(clazz.getDeclaredFields()).anyMatch(f -> !Modifier.isStatic(f.getModifiers()));
```
...might be a little more efficient.